File: files/fraser-coho-workshop/scripts/validate_sdp.py
"""Validate a draft, or pass --strict for the final SDP gate."""
import sys
import metasalmonpy as ms
from workshop import PACKAGE, check_preparation
check_preparation()
args = sys.argv[1:]
strict = "--strict" in args
paths = [arg for arg in args if arg != "--strict"]
validation = ms.validate_salmon_datapackage(paths[0] if paths else str(PACKAGE), require_iris=strict)
print(validation["issues"])
print(validation["semantic_validation"]["issues"])
if not validation["issues"].empty:
    raise SystemExit("Validation reported issues; resolve them before continuing.")
print("SDP validation passed. This does not establish scientific approval or EML readiness.")

File: files/fraser-coho-workshop/scripts/validate_sdp.R
# Chapter 4 onward: validate a draft, or use --strict for the final SDP gate.
library(metasalmon)
source("scripts/workshop.R")
check_preparation()
args <- commandArgs(trailingOnly = TRUE)
strict <- "--strict" %in% args
paths <- args[args != "--strict"]
path <- if (length(paths)) paths[[1]] else workshop_path
validation <- validate_salmon_datapackage(path, require_iris = strict)
print(validation$issues)
print(validation$semantic_validation$issues)
if (nrow(validation$issues)) stop("Validation reported issues; read and resolve them before continuing.")
message("SDP validation passed. This does not establish scientific approval or EML readiness.")
